calico_node/filesystem/startup.py

```diff
@@ -261,7 +261,7 @@ def main():
     # This is required because the confd templates expect to be able to fill in
     #  the IP by fetching it from the environment.
     with open('startup.env', 'w') as f:
-        f.write("IP=" + ip)
+        f.write("IP=" + str(ip))
```

Comment:
I would convert everything to handle IPAddresses but I have speed and memory concerns if we're constantly encapsulating strings in the larger objects. Maybe we can take the hit considering our python binary isn't focused on either of those currently...